mlrun/api/crud/artifacts.py

```diff
@@ -33,40 +33,78 @@ def store_artifact(
         self,
         db_session: sqlalchemy.orm.Session,
         key: str,
-        data: dict,
-        uid: str,
+        artifact: dict,
+        object_uid: str = None,
         tag: str = "latest",
         iter: int = 0,
-        project: str = mlrun.mlconf.default_project,
+        project: str = None,
+        producer_id: str = None,
     ):
         project = project or mlrun.mlconf.default_project
         # In case project is an empty string the setdefault won't catch it
-        if not data.setdefault("project", project):
-            data["project"] = project
+        if not artifact.setdefault("project", project):
+            artifact["project"] = project
 
-        if data["project"] != project:
+        if artifact["project"] != project:
             raise mlrun.errors.MLRunInvalidArgumentError(
-                f"Artifact with conflicting project name - {data['project']} while request project : {project}."
-                f"key={key}, uid={uid}, data={data}"
+                f"Artifact with conflicting project name - {artifact['project']} while request project : {project}."
+                f"key={key}, producer_id={producer_id}"
             )
-        mlrun.api.utils.singletons.db.get_db().store_artifact(
+        return mlrun.api.utils.singletons.db.get_db().store_artifact(
             db_session,
             key,
-            data,
-            uid,
+            artifact,
+            object_uid,
             iter,
             tag,
             project,
+            producer_id=producer_id,
         )
 
-    def get_artifact(
+    def create_artifact(
         self,
         db_session: sqlalchemy.orm.Session,
         key: str,
+        artifact: dict,
         tag: str = "latest",
         iter: int = 0,
+        tree: str = None,
```

Comment:
```suggestion
        producer_id: str = None,
```
make sure where you call it to use kwargs notation `producer_id=tree`